core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -133,72 +133,62 @@ class SYCLTeamMember {
     const unsigned int team_rank_ = team_rank();
 
     // First combine the values in the same subgroup
+#if defined(KOKKOS_ARCH_INTEL_GPU) || defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+    auto shuffle_combine = [&](int shift) {
+      if (vector_range * shift < sub_group_range) {
+        const value_type tmp = sg.shuffle_down(value, vector_range * shift);
+        if (team_rank_ + shift < team_size_) reducer.join(value, tmp);
+      }
+    };
+    shuffle_combine(1);
+    shuffle_combine(2);
+    shuffle_combine(4);
+    shuffle_combine(8);
+    shuffle_combine(16);
+    shuffle_combine(32);
+#else
     for (unsigned int shift = 1; vector_range * shift < sub_group_range;
          shift <<= 1) {
       const value_type tmp = sg.shuffle_down(value, vector_range * shift);
       if (team_rank_ + shift < team_size_) reducer.join(value, tmp);
     }
+#endif
     value = sg.shuffle(value, 0);
 
-    const auto n_subgroups = sg.get_group_range()[0];
+    const int n_subgroups = sg.get_group_range()[0];
     if (n_subgroups == 1) {
       reducer.reference() = value;
       return;
     }
 
-    // We need to chunk up the whole reduction because we might not have
-    // allocated enough memory.
-    const unsigned int maximum_work_range =
-        std::min<int>(m_team_reduce_size / sizeof(value_type), n_subgroups);
+    constexpr int step_width = 16;
+    auto tmp_alloc = sycl::ext::oneapi::group_local_memory_for_overwrite<
```

Comment:
We are just assuming that it's there for the oldest compiler version we support (which is in the CI). Otherwise, I'm not sure if it's actually used and we might not detect when this gets adopted in the SYCL standard since the macro definition might disappear then. We also are not very consistent when using other extensions (because some of them report not being usable when they are which might be debatable).